### PR TITLE
Adding "Unrealistic forces in Unity" test scene

### PR DESCRIPTION
**Pull request by chris24sahadeo**
bad_forces

### CODE REVIEW

File: minimal_test_cases/Assets/force_torque_issue/AddForce.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    private Transform _tipTransform;
+    
+    void Start()
+    {
+        // set force strength
+        if (forceStrength <= 0)
+        {
+            forceStrength = 5;
+            Debug.Log("Set defualt force: " + forceStrength);
+        }
```
Comment (vpradeep07) [inline minimal_test_cases/Assets/force_torque_issue/AddForce.cs]: Having defaults isn't really adding much towards conveying the problem at hand, and just results in additional code & logic that readers need to parse through.  I'd suggesting stripping this code out, along with the other defaults.
As an alternative, it's also reasonable to have error checking on the input values, and then print errors (or crash the app) if the values are not set as expected.

File: minimal_test_cases/Assets/force_torque_issue/AddForce.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        bool hit = Physics.Raycast(rayBeginInWorld, rayDirectionInWorld, out raycastHitInfo, RayDistance);
+        Debug.DrawRay(rayBeginInWorld, rayDirectionInWorld * RayDistance, Color.green, 1f, false);
+
+        // get hit info
+        Vector3 hitPointInWorld = raycastHitInfo.point;
+        Rigidbody hitObject = raycastHitInfo.rigidbody;
+
+        // apply force at hit point in direction of tip
+        Vector3 force = -rayDirectionInWorld.normalized * forceStrength;
+        hitObject.AddForceAtPosition(force, hitPointInWorld);
```
Comment (vpradeep07) [inline minimal_test_cases/Assets/force_torque_issue/AddForce.cs]: I was sometimes getting this error:
```
NullReferenceException: Object reference not set to an instance of an object
AddForce.FixedUpdate () (at Assets/force_torque_issue/AddForce.cs:55)
```

**Review by vpradeep07 — CHANGES_REQUESTED** [review]
Some comments inline.  Some meta points:
* Ideally, the unexpected behavior will be obvious, without the user interacting with the scene.  Here are 2 ways you could show this:
  * Idea 1: Add a script that moves the magnet in a sinusoidal linear motion
  * Idea 2: Create two instances of the current setup, side by side.  One will show the magnet working as expected (centered on the block), and the other will show the magnet not working as expected (near the edge of the block)
* Please update the game view camera to have a more useful viewpoint.
* Once this is merged (or about to be merged), please add an animated gif to the PR that shows what's happening in the scene.  This will help for future reference, so that folks don't have to fire up the entire project to see what's going on.

**Review by chris24sahadeo — COMMENTED** [review]
Changes made. Requesting review.

**Review by vpradeep07 — APPROVED** [review]
LGTM.  Thanks!

**Pull request merged**